llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -741,6 +747,36 @@ bool isTargetCompatibleWithModule(const std::string &Target,
   return true;
 }
 
+std::optional<module_split::ModuleDesc>
+makeDummy(module_split::ModuleDesc &MD) {
+  bool hasVirtualFunctions = false;
+  bool hasOptionalKernelFeatures = false;
+  for (Function &F : MD.getModule().functions()) {
+    if (F.hasFnAttribute("indirectly-callable"))
+      hasVirtualFunctions = true;
+    if (F.getMetadata("sycl_used_aspects"))
+      hasOptionalKernelFeatures = true;
+    if (hasVirtualFunctions && hasOptionalKernelFeatures)
+      break;
+  }
+  if (!hasVirtualFunctions || !hasOptionalKernelFeatures)
+    return {};
+
+  auto MDCopy = MD.clone();
+
+  for (Function &F : MDCopy.getModule().functions()) {
+    if (!F.hasFnAttribute("indirectly-callable"))
+      continue;
+
+    F.erase(F.begin(), F.end());
+    BasicBlock *newBB = BasicBlock::Create(F.getContext(), "entry", &F);
+    IRBuilder<> builder(newBB);
+    builder.CreateRetVoid();
```

Comment:
This may not be legal if a function has non-void return type. I suppose that we could do an alloca for return type, then do a load and then return that loaded value - even though it is not initialized, that code is not expected to be reachable anyways.

We could even insert something like `unreachable`, but I'm not sure how well it can be translated into SPIR-V, so for now it could be left in form of a simple `TODO`